contributors/design-proposals/standardization-of-development-process.md

```diff
@@ -0,0 +1,282 @@
+# Proposal to Standardize Development Process
+
+## Links
+```
+---
+Links:
+- Issues:
+  - [someIssueURL]()
+- PRs:
+  - https://github.com/kubernetes/community/pull/967
+- MailingListDiscussions:
+  - https://groups.google.com/forum/#!topic/kubernetes-dev/65A-3ULYPB0
+- Documentation:
+  - [someDocsLinkURL]()
+```
+
+## Responsible SIG(s)
+
+- SIG Release
+- SIG PM
+- SIG Architecture
+- SIG Governance
+- Technical Steering Committee
+
+## Summary
+
+A standardized development process for Kubernetes is proposed in order to:
+- provide a common structure for proposing changes to Kubernetes
```

Comment:
One more item:

- Ensure community participants are able to successfully drive changes to completion across one or more releases
  - Ensure stakeholders are adequately represented throughout

Which may overlap with above and below, but I think that the goal of the development process is to manage change across releases, not just within them, and releases are vehicles for features (rather than the other way around)

Reply:
More items in the email thread:
https://groups.google.com/forum/#!topic/kubernetes-sig-architecture/t-1EqeEoLPA

Reply:
I agree entirely @smarterclayton that releases are vehicles for features and not the other way around. I was trying to motivate a process which would tie together the continuous work of proposing, scoping, and implementing features, with how the results of that work would filter into user or developer facing documentation so that people can actually consume the new features.

Reply:
I hate to throw a spanner in the works so late in the process, but I have a concern/suggestion.  It seems that we're trying to address two mostly independent sets of concerns here:

1. Product and technical integrity
    1. Do we want this proposed thing (feature addition, enhancement, defect correction etc) to be applied to Kubernetes?
    2. If so, what exactly should it look like to users?
    3. What should the internal implementation look like?
    4. How do we ensure that it can be designed and implemented as quickly/efficiently as possible?

2. Progress tracking and communication.
    1. When will users be able to use it.
    2. How should users use it.

For in-SIG things, the answers to the above are fairly straightforward.  An individual SIG makes most of the decisions in relative isolation, and communicates via their meetings, notes, roadmap, release notes etc.

For cross-SIG things, it's clearly more complicated, and that's what we're trying to address here.

I would suggest that we first focus squarely on answering 1.1-1.4 above, and only then try to answer 2.1-2.2 within the above context.